doc/neps/nep-0047-array-api-standard.rst

```diff
@@ -0,0 +1,539 @@
+   libraries are not. Having one protocol per device isn't tenable, hence
+   device support is a must.
+2. Widespread support. DLPack has the widest adoption of all protocols, only
+   NumPy is missing support. And the experiences of other libraries with it
+   are positive. This contrasts with the protocols NumPy does support, which
+   are used very little - when other libraries want to interoperate with
+   NumPy, they typically use the (more limited, and NumPy-specific)
+   ``__array__`` protocol.
+
+Adding support for DLPack to NumPy entails:
+
+- Adding a ``ndarray.__dlpack__`` method
+- Adding a ``from_dlpack`` function, which takes as input an object
+  supporting ``__dlpack__``, and returns an ``ndarray``.
+
+DLPack is currently a ~200 LoC header, and is meant to be included directly, so
+no external dependency is needed. Implementation should be straightforward.
+
+
+Syntax for device support
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+NumPy itself is CPU-only, so it clearly doesn't have a need for device support.
+However, other libraries (e.g. TensorFlow, PyTorch, JAX, MXNet) support
+multiple types of devices: CPU, GPU, TPU, and more exotic hardware.
+To write portable code on systems with multiple devices, it's often necessary
+to create new arrays on the same device as some other array, or check that
+two arrays live on the same device. Hence syntax for that is needed.
+
+The array object will have a ``.device`` attribute which enables comparing
+devices of different arrays (they only should compare equal if both arrays are
+from the same library and it's the same hardware device). Furthermore,
+``device=`` keywords in array creation functions are needed. For example::
+
+    def empty(shape: Union[int, Tuple[int, ...]], /, *,
+              dtype: Optional[dtype] = None,
+              device: Optional[device] = None) -> array:
+        """
+        Array API compatible wrapper for :py:func:`np.empty <numpy.empty>`.
+        """
+        return np.empty(shape, dtype=dtype, device=device)
+
+The implementation for NumPy may be as simple as setting the device attribute to
+the string ``'cpu'`` and raising an exception if array creation functions
+encounter any other value.
+
+
+Dtypes and casting rules
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The supported dtypes in this namespace are boolean, 8/16/32/64-bit signed and
+unsigned integer, and 32/64-bit floating-point dtypes. These will be added to
+the namespace as dtype literals with the expected names (e.g., ``bool``,
+``uint16``, ``float64``).
+
+The most obvious omissions are the complex dtypes. The rationale for the lack
+of complex support in the first version of the array API standard is that several
+libraries (PyTorch, MXNet) are still in the process of adding support for
+complex dtypes. The next version of the standard is expected to include ``complex32``
+and ``complex64`` (see `this issue <https://github.com/data-apis/array-api/issues/102>`__
+for more details).
```

Comment:
I think you meant `complex64` and `complex128`?

Reply:
good catch, fixed now